lightningd/jsonrpc.c

```diff
@@ -509,12 +525,12 @@ static void parse_request(struct json_connection *jcon, const jsmntok_t tok[])
 	}
 
 	db_begin_transaction(jcon->ld->wallet->db);
-	cmd->dispatch(jcon->current, jcon->buffer, params);
+	cmd->dispatch(c, jcon->buffer, params);
 	db_commit_transaction(jcon->ld->wallet->db);
 
 	/* If they didn't complete it, they must call command_still_pending */
-	if (jcon->current)
-		assert(jcon->current->pending);
+	if (cmd_in_jcon(jcon, c))
```

Comment:
If triggering of `c` performed `command_success` or `command_fail*`, would it not already be freed by `json_done`?

Granted, `cmd_in_jcon` never actually accesses the `struct command`, but maybe mention that "we do not access `cmd` here as it might have gotten freed" as comment in `cmd_in_jcon`? We should not be adding commands while dispatching for a command, but in theory it would be possible to free the command, then allocate a new command, and then the system allocator returns the memory area that was recently freed, confusing things here (it will not confuse things here now as we do not add commands while dispatching for a command).